### PR TITLE
MINOR Removed copying storage libraries specifically as the task already copies them.

### PR DESCRIPTION
**Pull request by satishd**
- Removed copying storage libraries specifically as the task already copies them as part of `configurations.runtimeClasspath` in `releaseTarGz`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by satishd** [comment]
@junrao It is a minor change to address the issue you mentioned [here](https://github.com/apache/kafka/pull/10638#discussion_r627630009). Please take a look.

File: build.gradle [hunk 1]
```diff
@@ -1003,10 +1003,6 @@ project(':core') {
     from(project(':connect:mirror').configurations.runtimeClasspath) { into("libs/") }
     from(project(':connect:mirror-client').jar) { into("libs/") }
     from(project(':connect:mirror-client').configurations.runtimeClasspath) { into("libs/") }
-    from(project(':storage').jar) { into("libs/") }
-    from(project(':storage').configurations.runtimeClasspath) { into("libs/") }
-    from(project(':storage:api').jar) { into("libs/") }
-    from(project(':storage:api').configurations.runtimeClasspath) { into("libs/") }
```
Comment (ijuma) [inline build.gradle:1009]: Can you clarify why this is not needed?

Comment (satishd) [reply build.gradle:1009]: I observed that these libraries are copied as mentioned [here](https://github.com/apache/kafka/pull/10647/commits/1ad50e8634d56622e58a50c8328205b6db3f528d#diff-49a96e7eea8a94af862798a45174e6ac43eb4f8b4bd40759b5da63ba31ec3ef7R981). 

```
from(configurations.runtimeClasspath) { into("libs/") }
```

**Review by junrao — APPROVED** [review]
@satishd : Thanks for the PR. LGTM

File: build.gradle (hunk 1 above)
Comment (ijuma) [reply build.gradle:1009]: Makes sense, we only need to add modules that are not a core dependency already.

**Pull request merged**